g4f/Provider/HuggingChat.py

```diff
@@ -86,25 +86,36 @@ async def create_async_generator(
                 keys: list[int] = data[data[0]["messages"]]
                 message_keys: dict = data[keys[0]]
                 message_id: str = data[message_keys["id"]]
+
+            headers = {
+                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
+                "Content-Type": "multipart/form-data; boundary=----Web",
+                "Origin": "https://huggingface.co",
+            }
             options = {
                 "id": message_id,
                 "inputs": format_prompt(messages) if conversation is None else messages[-1]["content"],
                 "is_continue": False,
                 "is_retry": False,
-                "web_search": web_search
+                "web_search": web_search,
             }
-            async with session.post(f"{cls.url}/conversation/{conversation_id}", json=options) as response:
-                first_token = True
+            options_json=json.dumps(options)
+            body = (
+                    '------Web\r\n'
+                    'Content-Disposition: form-data; name="data"\r\n\r\n'
+                    f'{options_json}\r\n'
+                    '------Web--'
+            )
+
+            async with session.post(f"{cls.url}/conversation/{conversation_id}", headers=headers, data=body) as response:
                 async for line in response.content:
                     await raise_for_status(response)
                     line = json.loads(line)
                     if "type" not in line:
                         raise RuntimeError(f"Response: {line}")
                     elif line["type"] == "stream":
                         token = line["token"]
-                        if first_token:
-                            token = token.lstrip().replace('\u0000', '')
-                            first_token = False
+                        token = token.lstrip().replace('\u0000', ' ')
```

Comment:
Is it really necessary to use a whitespace here instead of an empty string?

Probably something is wrong because I'm getting lots of whitespace characters using the API

![image](https://github.com/xtekky/gpt4free/assets/35491968/a04311c3-38e0-41d1-b862-f83c02aaf739)

but, just replacing `' '` with `''` removes all spaces.

try this

```suggestion
                        token = token.replace('\u0000', '')
```

it works fine, at first glance

Reply:
Have few , more changes. Will send a new pull request tomorrow.

Reply:
if i replace  '  ' with ''. Then there is no space between the tokens.

So all the tokens are attached together when i tested it .

Reply:
Yes, I've mentioned that :)

have you tried removing the `lstrip()` call like in my comment above?

Reply:
Yaa, sorry.
I fixed that, its working fine so i deleted the comment

Reply:
Nice, thanks

Reply:
> ![image](https://private-user-images.githubusercontent.com/35491968/342905013-a04311c3-38e0-41d1-b862-f83c02aaf739.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzODcyMjgsIm5iZiI6MTcxOTM4NjkyOCwicGF0aCI6Ii8zNTQ5MTk2OC8zNDI5MDUwMTMtYTA0MzExYzMtMzhlMC00MWQxLWI4NjItZjgzYzAyYWFmNzM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI2VDA3Mjg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2MzhiMWQwN2QyODc0NGE5ZmI5YmQxOGFjODlkNzQzNjEwYWQ4ZGFkZmVlY2UxNGRlZmY0Y2ZjMmQzMzVlNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1MfKPs7MQ9jTx1JjYTAtx_OCJDYv8fRYkn3xwzpb2jc)
> 

In the image, how were you able to save and use previous conversations,

I added a new comment in your issue where i sent a SS of how its working for http://127.0.0.1:8080/chat/ & in huggingface chat feature in its website. It's unable to remember the previous messages as they are getting overridden.

Reply:
@Anurag-Varma IDK it just worked like that for me, I just used your version :D

I used your fix and it put the entire history in the first message.

That's why I didn't get what you meant, at first. Cause I didn't have that problem, for some reason

But since xtekky has fixed the issue, it doesn't really matter anymore, I suppose. But if this problem is still reproducing for you (I haven't checked the new fix yet), maybe you should open a new issue on it. Try using the clean version from the repository just in case